src/button-group/internal.tsx

```diff
@@ -0,0 +1,297 @@
+// Copyright Amazon.com, Inc. or its affiliates. All Rights Reserved.
+// SPDX-License-Identifier: Apache-2.0
+import React, { useImperativeHandle, useRef, forwardRef, useEffect, useState, useMemo } from 'react';
+import { getBaseProps } from '../internal/base-component';
+import { ButtonGroupProps, InternalButtonGroupProps } from './interfaces';
+import { ButtonProps } from '../button/interfaces';
+import { ClickDetail, fireCancelableEvent } from '../internal/events/index.js';
+import {
+  SingleTabStopNavigationAPI,
+  SingleTabStopNavigationProvider,
+} from '../internal/context/single-tab-stop-navigation-context';
+import { useMergeRefs } from '../internal/hooks/use-merge-refs';
+import { KeyCode } from '../internal/keycode';
+import { hasModifierKeys } from '../internal/events';
+import { circleIndex } from '../internal/utils/circle-index';
+import { getAllFocusables } from '../internal/components/focus-lock/utils';
+import { nodeBelongs } from '../internal/utils/node-belongs';
+import { warnOnce } from '@cloudscape-design/component-toolkit/internal';
+import ItemElement from './item-element.js';
+import testUtilStyles from './test-classes/styles.css.js';
+import handleKey from '../internal/utils/handle-key';
+import Tooltip from '../internal/components/tooltip/index.js';
+import LiveRegion from '../internal/components/live-region/index.js';
+import clsx from 'clsx';
+import styles from './styles.css.js';
+
+const InternalButtonGroup = forwardRef(
+  (
+    {
+      items = [],
+      onItemClick,
+      ariaLabel,
+      dropdownExpandToViewport,
+      __internalRootRef = null,
+      ...props
+    }: InternalButtonGroupProps,
+    ref: React.Ref<ButtonGroupProps.Ref>
+  ) => {
+    const baseProps = getBaseProps(props);
+    const [focusedItemId, setFocusedItemId] = useState<null | string>(null);
+    const navigationAPI = useRef<SingleTabStopNavigationAPI>(null);
+    const containerObjectRef = useRef<HTMLDivElement>(null);
+    const containerRef = useMergeRefs(containerObjectRef, __internalRootRef);
+    const itemsRef = useRef<Record<string, ButtonProps.Ref | null>>({});
+    const itemWrappersRef = useRef<Record<string, HTMLDivElement | null>>({});
+    const [tooltipItemId, setTooltipItemId] = useState<string | null>(null);
+    const [isFeedbackTooltip, setIsFeedbackTooltip] = useState(false);
+    const allItems = useMemo(
```

Comment:
nit: I'd call the variable itemsById or similar to emphasise it is a dictionary type

Reply:
Renamed